modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-action/ActionTypeReassignment.js

```diff
@@ -9,58 +9,59 @@
  * distribution rights of the Software.
  */
 
-import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
 
-const ActionTypeReassignment = () => {
-	const [reassignmentSections, setReassignmentSections] = useState([
-		{identifier: `${Date.now()}-0`},
-	]);
+import {DiagramBuilderContext} from '../../../../../DiagramBuilderContext';
+import AssetCreator from '../select-reassignment/AssetCreator';
+import ResourceActions from '../select-reassignment/ResourceActions';
+import Role from '../select-reassignment/Role';
+import RoleType from '../select-reassignment/RoleType';
+import ScriptedAssignment from '../select-reassignment/ScriptedReassignment';
+import {SelectReassignment} from '../select-reassignment/SelectReassignment';
+import User from '../select-reassignment/User';
 
-	const deleteSection = (identifier) => {
-		setReassignmentSections((prevSections) => {
-			const newSections = prevSections.filter(
-				(prevSection) => prevSection.identifier !== identifier
-			);
+const assignmentSectionComponents = {
+	assetCreator: AssetCreator,
+	resourceActions: ResourceActions,
+	roleId: Role,
+	roleType: RoleType,
+	scriptedAssignment: ScriptedAssignment,
+	user: User,
+};
+
+const ActionTypeReassignment = (props) => {
+	const {selectedItem} = useContext(DiagramBuilderContext);
+	const assignmentType = selectedItem?.data?.assignments;
+	const [section, setSection] = useState(assignmentType || 'assetCreator');
+	const [sections, setSections] = useState([{identifier: `${Date.now()}-0`}]);
+	const ReassignmentSectionComponent = assignmentSectionComponents[section];
 
-			return newSections;
-		});
-	};
+	const [scriptSections] = useState([{identifier: `${Date.now()}-0`}]);
 
-	return reassignmentSections.map(({identifier}) => {
+	return scriptSections.map(({identifier}) => {
```

Comment:
@eliasantosf I don't understand why this state is named `scriptSections` instead of `reassignmentSections`